### PR TITLE
ltp: Add preliminary version of LTP 20190930

### PR DESCRIPTION
**Pull request by mrchapp**
All patches have been refreshed. These two were somewhat updated:
* 0004-build-Add-option-to-select-libc-implementation.patch
* 0020-getdents-define-getdents-getdents64-only-for-glibc.patch

This backported patch was dropped as it is now included by the released version:
* overcommit_memory-update-for-mm-fix-false-positive-O.patch

This is considered preliminary in that it's not ready to be upstreamed, as it's still missing a few patches for MUSL. :(

### CODE REVIEW

File: recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch [hunk 1]
```diff
@@ -39,18 +39,19 @@ index 297f8e7..906b280 100644
...
-index 50a12fa..4f1987f 100644
+index 3319b3163..94ea362c8 100644
 --- a/testcases/kernel/Makefile
 +++ b/testcases/kernel/Makefile
-@@ -49,12 +49,15 @@ SUBDIRS			+= connectors \
+@@ -49,13 +49,16 @@ SUBDIRS			+= connectors \
  			   logging \
  			   mem \
  			   numa \
 -			   pty \
```
Comment (roxell) [inline recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch:50]: where did the pty test go?

File: recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch [hunk 2]
```diff
@@ -39,18 +39,19 @@ index 297f8e7..906b280 100644
...
  			   sched \
  			   security \
- 			   timers \
```
Comment (roxell) [inline recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch:52]: where did the timers tests go?

**Review by roxell — COMMENTED** [review]
so it was only 0004-build-add-option-to-select* and 0020-getdents-defin-getdents-* that had problematicish merge conflicts ?

Great that we were able to remove the overcommit_memory-update-for-mm-* patch.

How much are missing for MUSL?
Can we send it to OE and CC Petr the LTP maintainer too speed up that part?

File: recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch (hunk 1 above)
Comment (mrchapp) [reply recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch:50]: It was made conditional a few lines below.

File: recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch (hunk 2 above)
Comment (mrchapp) [reply recipes-extended/ltp/ltp/0004-build-Add-option-to-select-libc-implementation.patch:52]: Was removed by LTP in 0051eab232e7f777f435945b1aba8911f203d65b.

**Comment by mrchapp** [comment]
> so it was only 0004-build-add-option-to-select* and 0020-getdents-defin-getdents-* that had problematicish merge conflicts ?

It was a simple rebase. Those were the only two that had substantial changes in the interdiff. No conflict at all. All other patches would have applied as-is with a little fuzzing automatically.

> Great that we were able to remove the overcommit_memory-update-for-mm-* patch.

Yes, we took it from LTP master, so it's there now.

> How much are missing for MUSL?

IIRC, there's still one difficult patch.

> Can we send it to OE and CC Petr the LTP maintainer too speed up that part?

Not sure. For certain, that patch could not be taken by OE as it would break MUSL.

**Comment by roxell** [comment]
On Wed, Oct 9, 2019 at 9:46 PM Daniel Díaz <notifications@github.com> wrote:
>
> so it was only 0004-build-add-option-to-select* and 0020-getdents-defin-getdents-* that had problematicish merge conflicts ?
>
> It was a simple rebase. Those were the only two that had substantial changes in the interdiff. No conflict at all. All other patches would have applied as-is with a little fuzzing automatically.
>
> Great that we were able to remove the overcommit_memory-update-for-mm-* patch.
>
> Yes, we took it from LTP master, so it's there now.
>
> How much are missing for MUSL?
>
> IIRC, there's still one difficult patch.
>
> Can we send it to OE and CC Petr the LTP maintainer too speed up that part?
>
> Not sure. For certain, that patch could not be taken by OE as it would break MUSL.

What I meant was would it be OK to send it out to OE as an RFC or
something and ask Petr  to take a look at it?

worth a shot or?

**Pull request merged**